daemon/cmd/daemon_main.go

```diff
@@ -804,6 +804,15 @@ func init() {
 	flags.String(option.HubbleListenAddress, "", `An additional address for Hubble server to listen to, e.g. ":4244"`)
 	option.BindEnv(option.HubbleListenAddress)
 
+	flags.Bool(option.HubbleAllowInsecure, false, "Allow Hubble server to run on the given listen address without TLS.")
```

Comment:
I think this PR is too big to be backported to v1.8 and has potential for breakages if we are not careful enough. I feel it's probably safer to keep this feature for v1.9. What do you think?

Reply:
As discussed offline, we decided not to backport this to v1.8. However, my comment still applies fully, since the default behavior (mTLS on Hubble Server) still changes from 1.8 to 1.9. So we still need the upgrade notes and we still need to `upgradeCompatibility` section in the configmap chart mentioned below.

Reply:
This was discussed offline. It turns out there are multiple options we could choose for a non-disruptive upgrade path.
I have created a release blocker issue to ensure this problem is addressed before Cilium v1.9 is released: #13068.
Every solution other than simply documenting a way to upgrade without breakage are non-trivial and would thus be addressed in a follow-up pull request.